app/models/category.rb

```diff
@@ -0,0 +1,14 @@
+class Category
+  include Mongoid::Document
+
+  has_many :syntactical_words, class_name: "Word", inverse_of: :syntactical_category, :autosave => true
+  has_and_belongs_to_many :semantical_words, class_name: "Word", inverse_of: :semantical_categories, :autosave => true
```

Comment:
Why has a category both semantical and syntactical words. If I recall correctly a category is either semantic or syntactic and hence its words are implicitly syntactic or semantic but never both.